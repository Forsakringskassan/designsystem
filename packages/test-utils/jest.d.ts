export * from "./dist/types/jest";

declare global {
    namespace jest {
        interface Matchers<R> {
            /**
             * Expects element to have focus.
             */
            toHaveFocus(): R;
        }
    }
}
