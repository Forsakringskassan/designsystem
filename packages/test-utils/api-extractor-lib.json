{
    "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
    "extends": "../../api-extractor.json",
    "mainEntryPointFilePath": "<projectFolder>/temp/types/lib.d.ts",
    "apiReport": {
        "enabled": false
    },
    "dtsRollup": {
        "publicTrimmedFilePath": "<projectFolder>/dist/types/lib.d.ts"
    }
}
