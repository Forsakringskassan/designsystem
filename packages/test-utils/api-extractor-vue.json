{
    "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
    "extends": "../../api-extractor.json",
    "mainEntryPointFilePath": "<projectFolder>/temp/types/vue/index.d.ts",
    "apiReport": {
        "enabled": false
    },
    "dtsRollup": {
        "publicTrimmedFilePath": "<projectFolder>/dist/types/vue.d.ts"
    }
}
