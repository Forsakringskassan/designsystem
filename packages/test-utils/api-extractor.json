{
    "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
    "extends": "../../api-extractor.json",
    "tsdocMetadata": {
        "tsdocMetadataFilePath": "<projectFolder>/dist/types/tsdoc-metadata.json"
    },
    "dtsRollup": {
        "enabled": false
    }
}
