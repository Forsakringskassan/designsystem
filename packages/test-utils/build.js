const fs = require("node:fs/promises");
const path = require("node:path");
const esbuild = require("esbuild");

const type = {
    cjs: "commonjs",
    esm: "module",
};

async function build() {
    for (const format of ["cjs", "esm"]) {
        const outdir = `dist/${format}`;
        const common = {
            logLevel: "info",
            sourcemap: true,
            external: ["vue", "@fkui/*"],
            bundle: true,
            platform: "node",
            format,
            target: "node20",
            outdir,
        };
        await esbuild.build({
            ...common,
            entryPoints: ["src/lib.ts", "src/jest.ts"],
        });
        await esbuild.build({
            ...common,
            entryPoints: [{ in: "src/vue/index.ts", out: "vue" }],
            define: {
                "process.env.NODE_ENV": JSON.stringify("development"),
            },
        });
        const pkg = JSON.stringify({ type: type[format] }, null, 2);
        await fs.writeFile(path.join(outdir, "package.json"), pkg, "utf-8");
    }
}

build().catch((err) => {
    console.error(err);
});
