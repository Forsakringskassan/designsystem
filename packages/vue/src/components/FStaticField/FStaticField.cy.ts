import { FStaticFieldPageObject } from "../../cypress";
import FStaticFieldInput from "./examples/FStaticFieldInput.vue";
import FStaticFieldTooltipDescription from "./examples/FStaticFieldTooltipDescription.vue";

describe("FStaticField", () => {
    it("static field with textfield: should provide a page object that can access any necessary elements", () => {
        cy.mount(FStaticFieldInput);
        const staticField = new FStaticFieldPageObject(
            "[data-test] .output-field",
        );
        staticField.label.el().eq(0).should("have.trimmedText", "Beskrivning");
        staticField.tooltip.el().should("not.exist");
        staticField
            .body()
            .should("have.trimmedText", "En liten statisk beskrivning.");
    });

    it("static field with tooltip: should provide a page object that can access any necessary elements", () => {
        cy.mount(FStaticFieldTooltipDescription);
        const staticFieldWithTooltip = new FStaticFieldPageObject(
            "[data-test] .output-field",
        );
        staticFieldWithTooltip.label.el().should("contain.text", "Etikett");
        staticFieldWithTooltip.tooltip.el().should("not.be.visible");
        staticFieldWithTooltip.tooltip.iButton().click();
        staticFieldWithTooltip.tooltip.el().should("be.visible");
        staticFieldWithTooltip
            .body()
            .should("have.trimmedText", "En liten statisk text.");
    });
});
