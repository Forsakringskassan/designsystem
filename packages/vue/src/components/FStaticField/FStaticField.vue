<template>
    <div class="output-field">
        <f-label>
            <template #default>
                <!-- @slot Slot for label content. -->
                <slot name="label"></slot>
            </template>
            <template v-if="$slots.tooltip" #tooltip>
                <!-- @slot Slot primarily for tooltip. -->
                <slot name="tooltip"></slot>
            </template>

            <template #description="{ descriptionClass, discreteDescriptionClass }">
                <!--
                    @slot Optional slot for description. See [FLabel](#/Components/FLabel) for details.
                    @binding {string[]} descriptionClass CSS classes for primary description content.
                    @binding {string[]} discreteDescriptionClass CSS classes for format description.
                -->
                <slot name="description" v-bind="{ descriptionClass, discreteDescriptionClass }"></slot>
            </template>
        </f-label>
        <p class="output-field__output">
            <!-- @slot Slot for text content. -->
            <slot name="default"></slot>
        </p>
    </div>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import { FLabel } from "../FLabel";

export default defineComponent({
    name: "FStaticField",
    components: {
        FLabel,
    },
});
</script>
