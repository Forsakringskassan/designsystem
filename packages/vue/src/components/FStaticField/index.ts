export { default as FStaticField } from "./FStaticField.vue";
