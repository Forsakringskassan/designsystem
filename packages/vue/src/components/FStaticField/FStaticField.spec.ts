import { VueWrapper, mount } from "@vue/test-utils";
import FStaticField from "./FStaticField.vue";

function createWrapper({ slots = {} } = {}): VueWrapper {
    return mount(FStaticField, {
        slots: { ...slots },
    });
}

describe("snapshots", () => {
    it("should match snapshot with span and p", () => {
        const wrapper = createWrapper({
            slots: { label: "Heading", default: "En liten text" },
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    it("should match snapshot with span, p and tooltip", () => {
        const wrapper = createWrapper({
            slots: {
                label: "Heading",
                tooltip: "TOOLTIP",
                default: "En liten text",
            },
        });

        expect(wrapper.element).toMatchSnapshot();
    });
});
