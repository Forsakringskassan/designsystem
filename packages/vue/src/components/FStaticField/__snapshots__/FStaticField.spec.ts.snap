// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots should match snapshot with span and p 1`] = `
<div
  class="output-field"
>
  <label
    class="label"
  >
    
    
    Heading
    
    
    
    
    <!--v-if-->
  </label>
  <p
    class="output-field__output"
  >
    
    En liten text
    
  </p>
</div>
`;

exports[`snapshots should match snapshot with span, p and tooltip 1`] = `
<div
  class="output-field"
>
  <div>
    <div
      class="tooltip-before"
    >
      <label
        class="label tooltip-before__label"
      >
        
        
        Heading
        
        
      </label>
    </div>
    
    
    TOOLTIP
    
    
    <!--v-if-->
  </div>
  <p
    class="output-field__output"
  >
    
    En liten text
    
  </p>
</div>
`;
